### PR TITLE
Borer Bugfixes

### LINKED ISSUE #6629: Decapitation causes ghosting out of borer (and maybe other stuff too?)
(WEB REPORT BY: assdad211 REMOTE: 158.69.55.2:7777)

> Revision 086f427dd6269110ee5243d502c0567dc9f11f3f
> 
> General description of the issue

I was playing as a cortical borer inside the clown. I had a dead body lying on the station, my former body. A traitor with an esword decapitated that dead body and I was instantly ghosted out of the borer/clown and teleported to the site of my former body which just got decapitated.

> What you expected to happen
> 
> What actually happened
> 
> Steps to reproduce if possible

### LINKED ISSUE #10911: Borer thought-speech inaudible as ghost w/o GhostEars but inrange
(WEB REPORT BY: wizardcrying REMOTE: 172.93.109.202:7777)
Revision 0d91b25673f35182d69c6e36175ad143f2e1db89

It doesn't seem that it's possible to hear borers inside of anyone without enabling GhostEars. Even if you're within earshot of the host, thought-speech is still inaudible.

### LINKED ISSUE #11020: Borer in slime
(WEB REPORT BY: repertor REMOTE: 172.93.109.202:7777)

> Revision c6609a726a394fddd0f2ec974c28fd1c77810b02 
> 
> General description of the issue
> borer is kill when slime becomes adult
> 
> What you expected to happen
> get transferred to new mob
> 
> What actually happened
> i dead pls restart
> 
> Steps to reproduce if possible
> be in slime as borer while others feed it monkeys

### LINKED ISSUE #11114: Borers die to Xeno gibbing, but not any other sort of gibbing
It's weird, a larva bursting out of a person will force ghost the borer, but any other gibbing will have them reside in the head object. Haven't tried with other limbs.

### LINKED ISSUE #11385: Even in death I flesh hook
(WEB REPORT BY: repertor REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)486f9b1d5f29afe33bf647e74fe113b9e0b44da8 
> 
> General description of the issue
> hook work in death
> 
> What you expected to happen
> hook stop work
> 
> What actually happened
> HOOK ONTO EVERYTHING ONLY CHOPPING THE ARM OFF CAN STOP ME
> 
> Steps to reproduce if possible
> die as extend arm borer, still hook

### LINKED ISSUE #11897: 2x attack bug abuse
#### Description of issue

You can attack with the same weapon, or a different weapon, TWICE REALLY FAST. Unsure about the later, but do know the same weapon gets 2 uses.
#### Difference between expected and actual behavior

Expected: normal attacking
Happening: you and your borer attack
#### Steps to reproduce
1. get an arm borer
2. get it to be hook shot module
3. give it a weapon of your choice
4. attack with your weapon while your borer is attacking with it
5. you will notice you and your borer sharing your arm have two separate cooldowns for attacking, so you can attack 2x as fast

### PR DESCRIPTION
**Pull request by Shadowmech88**
Borers are now properly placed into severed limbs.
Fixes #6629
Fixes #11114

Being turned into a monkey no longer causes head borers to drop.

Fixes #10911
Fixes #11020
Fixes #11385

The fleshshot is no longer capable of being dropped through any means.

As far as _adjacent_ attacks go, arm borers' attack delays are now tied directly to their hosts (and, by extension, the other arm borer's).
Fixes #11897

What I did _should_ fix #10911 and #11897, but I can't really test these alone.
Every other fix has been tested.

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -611,26 +610,33 @@
...
-	for(var/mob/M in src)//mobs, all of them
-		if(M == B)
-			continue
-		if(M in src.stomach_contents)
-			src.stomach_contents.Remove(M)
-		M.forceMove(target)
+//	var/mob/living/simple_animal/borer/B = has_brain_worms()
+	var/list/borer_list = get_brain_worms()
+	for(var/mob/living/simple_animal/borer/B in borer_list)
+		for(var/mob/M in src)//mobs, all of them
```
Comment (Intigracy) [inline code/modules/mob/living/carbon/carbon.dm]: Untab this.

**Review by Intigracy — CHANGES_REQUESTED** [review]
Tabbing this the way you are is breaking eating mobs being properly cleaned up.

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (Intigracy) [reply code/modules/mob/living/carbon/carbon.dm]: Unless this somehow works even when no borers are in them.

Comment (Shadowmech88) [reply code/modules/mob/living/carbon/carbon.dm]: Alright, it should function the same way it did before, now.

**Review by Intigracy — APPROVED** [review]

**Comment by Shadowmech88** [comment]
Added a new fix for #11897.
It should work in theory, but I can't test it myself.

**Comment by Shadowmech88** [comment]
Compile failure fixed.

**Pull request merged**